### ISSUE TITLE
Is this crate unmaintained?

### ISSUE THREAD

**Issue by MarijnS95** [gfx-rs/metal-rs#313]
Even though a release has been published just two days ago, lots of perfectly good PRs appear to be ignored, making it pretty much impossible to use this crate for anything more than the absolute basics. Such PRs include both soundness fixes and exposure of missing API surface.

Is there anything that can be done to pick up the pace and slack on the pull request backlog? This is important for us as we're working towards shipping crates and applications based on `metal-rs`. We'd hate to have to desert to publishing a fork with all the missing fixes and features because of this, but that looks to become our only option.

Note also that this lack of activity is actively discouraging further contributions. We have a lot more patches we'd like to upstream, but a big dependency stack in PRs is a nontrivial burden to handle - something that doesn't have to happen when PRs are actively reviewed and merged.

Thanks!

**Comment by MarijnS95** [comment]
@grovesNL thanks for the update!

`objc2` has been in consideration, but we have enough `metal-rs` code that migrating over is no small feat - even though I recognize it's likely the way forward. As said in the `wgpu` Matrix channel that I just found, it feels a bit harsh to request/force maintenance when a clear - autogenerated and maintained - alternative already exists.

It'd be nice to go over open PRs (and we'll open more for our backlog), and on the side I'll keep chipping away at seeing what `objc2` can do for us.

**Comment by MarijnS95** [comment]
We have gone through a lot of effort and trouble to add missing API bits left and right to `metal-rs` here (and there's much more left), but eventually bit the bullet and migrated to `objc2` entirely which has been an otherworldly productivity boost for us.

While I could ramble on about much improved documentation, more clear types (`Retained<ProtocolObject<dyn T>>` and variants) and more ergonomic function definitions, the number one win in `objc2` was having day-one access to new Metal 4 API, "just a generator rerun away" - and the confidence that the generated API bindings were actually correct.

Hence closing my issue as "not planned"; you all seem to maintaining `metal-rs` after all, but I can't state often enough how much you're missing out on from `objc2`.

**Comment by MarijnS95** [comment]
To me, adding new or previously-"missing" API such as being able to query for Metal 4 support (in that range you linked) is stretching the definition of "maintenance mode" (without adding any MTL4 API surface).

I can only imagine that sets a precedent for external contributors who might not be aware of this issue to start PR'ing `MTL4` API surface; where will the line be drawn?

**Comment by ErichDonGubler** [comment]
I'll let @inner-daemons and @cwfitzgerald answer about detection of `Metal4`. I suspect it was done because it was easy, but I'm not sure if or how wgpu actually needed it (which is the only reason I think we'd justify doing that, at this point).

**Comment by inner-daemons** [comment]
Yeah, I think Ive gotten like 2 PRs on this thing in. Both were very simple and necessary for something wgpu uses. I put up a PR and asked in our matrix for a quick review because I needed it, and that happened.

I can't really speak to other use cases. But certainly no one is scouring all the PRs here regularly trying to merge them. You'd have to bring it up with maintainers if you wanted to get stuff in.

And yeah we are trying to switch to objc2. I don't know why other non wgpu apps would want to use this.
